samples/rtl/axpy_double_pump.py

```diff
@@ -0,0 +1,456 @@
+import dace
+from dace.transformation.dataflow import StreamingMemory
+from dace.transformation.interstate import FPGATransformState
+from dace.transformation.dataflow import TrivialMapElimination
+import numpy as np
+
+# add symbol
+N = dace.symbol('N')
+
+# add sdfg
+sdfg = dace.SDFG('axpy_double_pump')
+
+# add state
+state = sdfg.add_state('device_state')
+
+# add parametr
+veclen = 2
+sdfg.add_constant('VECLEN', veclen)
+sdfg.add_constant('DATA_WIDTH', 64)
+sdfg.add_constant('WORD_WIDTH', 32)
+sdfg.add_constant('RATIO', 2)
+
+# add arrays
+sdfg.add_scalar('a', dtype=dace.float32, storage=dace.StorageType.FPGA_Global)
+sdfg.add_array('x', [N // veclen],
+               dtype=dace.vector(dace.float32, veclen),
+               storage=dace.StorageType.CPU_Heap)
+sdfg.add_array('y', [N // veclen],
+               dtype=dace.vector(dace.float32, veclen),
+               storage=dace.StorageType.CPU_Heap)
+sdfg.add_array('result', [N // veclen],
+               dtype=dace.vector(dace.float32, veclen),
+               storage=dace.StorageType.CPU_Heap)
+sdfg.add_array('fpga_x', [N // veclen],
+               dtype=dace.vector(dace.float32, veclen),
+               transient=True,
+               storage=dace.StorageType.FPGA_Global)
+sdfg.add_array('fpga_y', [N // veclen],
+               dtype=dace.vector(dace.float32, veclen),
+               transient=True,
+               storage=dace.StorageType.FPGA_Global)
+sdfg.add_array('fpga_result', [N // veclen],
+               dtype=dace.vector(dace.float32, veclen),
+               transient=True,
+               storage=dace.StorageType.FPGA_Global)
+
+# add streams
+sdfg.add_stream('x_stream',
+                buffer_size=32,
+                dtype=dace.vector(dace.float32, veclen),
+                transient=True,
+                storage=dace.StorageType.FPGA_Local)
+sdfg.add_stream('y_stream',
+                buffer_size=32,
+                dtype=dace.vector(dace.float32, veclen),
+                transient=True,
+                storage=dace.StorageType.FPGA_Local)
+sdfg.add_stream('result_stream',
+                buffer_size=32,
+                dtype=dace.vector(dace.float32, veclen),
+                transient=True,
+                storage=dace.StorageType.FPGA_Local)
+
+# add custom rtl tasklet
+rtl_tasklet = state.add_tasklet(name='rtl_tasklet',
+                                inputs={'a_in', 'x_in', 'y_in'},
+                                outputs={'result_out'},
+                                code='''
+    assign ap_done = 1; // free-running
+    wire clk_300;
+    wire clk_600;
+    wire rstn_300;
+    wire rstn_600;
+
+    clk_wiz_0 clk_wiz_inst (
+        .clk_in1(ap_aclk),
+        .clk_out1(clk_300),
+        .clk_out2(clk_600)
+    );
+
+    rst_clk_wiz rst_clk_wiz_300 (
```

Comment:
Could you add a rough ASCII block diagram somewhere to show what you're doing?

Reply:
Yes, working on it!